matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -287,7 +287,27 @@ impl<C: BuildablePlurality> WebRtcSocketBuilder<C> {
             .map(|(rx, tx)| Some(WebRtcChannel { rx, tx }))
             .collect();
 
-        let (id_tx, id_rx) = crossbeam_channel::bounded(1);
+        let (id_tx, id_rx) = futures_channel::oneshot::channel();
+
+        let socket_fut = run_socket(
+            id_tx,
+            self.config,
+            peer_messages_out_rx,
+            peer_state_tx,
+            messages_from_peers_tx,
+        )
+        .map(|f| {
+            f.map_err(|e| match e {
+                SignalingError::Undeliverable(_) => Error::Disconnected,
+                SignalingError::StreamExhausted => Error::Disconnected,
+                SignalingError::UnknownFormat => Error::Communication(e),
+                SignalingError::ConnectionFailed(_) => Error::ConnectionFailed,
+                SignalingError::Socket(_) => Error::Disconnected,
+                SignalingError::Packet(_) => Error::Disconnected,
+                #[cfg(target_arch = "wasm32")]
+                SignalingError::JsPacket(_) => Error::Disconnected,
+            })
```

Comment:
I'd rather see this implemented as `From<SignalingError> for Error` (or perhaps `Into`). Also, are we happy discarding the inner info or should we make `Disconnected` hold `Box<dyn<Error>>`?

Reply:
I attached a `source` field to each error to account, so no information is dropped. Good?
@johanhelsing

Reply:
`source` does the job nicely. Any thoughts on `From` vs the current match on `SignalingError`?

Reply:
Can't do From since the errors each hold the same error type, which would cause conflicting `From` impls.

Reply:
Ah, gotcha, didn't clock that one